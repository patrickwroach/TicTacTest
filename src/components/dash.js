import React, {Component} from 'react';
import {findWinner} from '../services/game-engine';



export class Dash extends React.Component {
     render() {  
        let message ='';  
        const win = findWinner(this.props.grid.squares);
        const turnNumber = this.props.turnCounter + 1;
        if (this.props.turnCounter===0){
           message = "First move goes to X!" ;    
        } else{
            if (win) {
            message = 'We have a winner, congrats ' + win + '!';
            } else {
                if(this.props.turnCounter<9){
                    message =  "It's turn " +  turnNumber + " and " + this.props.turn + " is up.";
                }else{
                    message="Cat's game."
                }

        }
    }

      return (
        <div className="dash section">
        
            <h2> 
            {message}
            </h2>
            <button onClick={this.props.newGame}>
               <img src="https://flaticons.net/gd/makefg.php?i=icons/Application/Command-Reset.png&amp;r=255&amp;g=255&amp;b=255" />
            </button>
         
        </div>
      );
    }
  }
  