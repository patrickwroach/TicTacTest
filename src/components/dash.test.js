import React from 'react';
import { shallow } from 'enzyme';
import {Dash} from './dash';

describe('Dash', () => {    

    const newGame = jest.fn();  
    const grid = {
        squares: Array(9).fill(null)
      };

  it('It should render correctly in "debug" mode', () => {  

    const component = shallow(<Dash 
        debug grid={grid}
         />);  
    expect(component).toMatchSnapshot();
  });

  it('refresh button click should fire passed function', () => {
    const component = shallow(<Dash grid={grid} newGame ={newGame} />);
    component
      .find('button#refresh')
      .simulate('click');
    expect(newGame).toHaveBeenCalled();
  });



 
});


