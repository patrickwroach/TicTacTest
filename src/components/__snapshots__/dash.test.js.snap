// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dash It should render correctly in "debug" mode 1`] = `
<div
  className="dash section"
>
  <h2>
    Cat's game.
  </h2>
  <button
    id="refresh"
  >
    <img
      src="https://flaticons.net/gd/makefg.php?i=icons/Application/Command-Reset.png&r=255&g=255&b=255"
    />
  </button>
</div>
`;
